src/client/rest/RESTMethods.js

```diff
@@ -136,7 +137,14 @@ class RESTMethods {
       );
   }
 
-  bulkDeleteMessages(channel, messages) {
+  bulkDeleteMessages(channel, messages, filterOld) {
+    if (filterOld) {
+      for (const i in messages) {
```

Comment:
This should be simplified down to a single `filter()` call, so that the original array isn't modified.